math/include/mainframe/math/vector3.h

```diff
@@ -65,7 +65,7 @@ namespace mainframe {
 				return {x, z, y};
 			}
 
-			VecType lerp(const VecType& other, NumberType timestep) const {
+			VecType lerp(const VecType& other, float timestep) const {
```

Comment:
this should be NumberType, if it's a vector of doubles, then this is invalid. NumberType is the template for it